src/NodeServices/Validity.cs

```diff
@@ -41,12 +41,6 @@ public NodeInternalException(string message) : base(message)
         public NodeInternalException(string message, Exception inner) : base(message, inner)
         {
         }
-
-        protected NodeInternalException(
-            SerializationInfo info,
-            StreamingContext context) : base(info, context)
-        {
-        }
```

Comment:
not used, but causes error because of BinaryFormatter (IFormatter) references in the base class.

Reply:
I think you can also remove the `[Serializable]` attribute, is this type used at all?

Reply:
It is used in 2 `Validity.Assert` functions

Reply:
I am not sure where the exceptions bubble up. Maybe they do get serialized somewhere...